src/control/server/engine/config.go

```diff
@@ -342,7 +345,53 @@ func (c *Config) Validate() error {
 	if err := ValidateLogSubsystems(subsystems); err != nil {
 		return errors.Wrap(err, "validate engine log subsystems")
 	}
-
+	return c.ValidateAndAdjustPMDKEnvVar()
+}
+
+// Ensure proper environment variables for PMDK w/ NDCTL enabled based on
+// the actual configuration of the storage class.
+func (c *Config) ValidateAndAdjustPMDKEnvVar() error {
+	if len(c.Storage.Tiers) > 0 {
+		pmemobj_conf_str, pmemobj_conf_err := c.GetEnvVar("PMEMOBJ_CONF")
+		if c.Storage.Tiers[0].Class == storage.ClassDcpm {
+			// Ensure 18KiB ABT stack size for an engine with DCPM storage class.
+			stacksize_str, err := c.GetEnvVar("ABT_THREAD_STACKSIZE")
+			if err != nil {
+				fmt.Printf("env_var ABT_THREAD_STACKSIZE set to %d for 'dcpm' storage class\n",
+					MIN_ABT_THREAD_STACKSIZE_FOR_DCPM)
+				c.EnvVars = append(c.EnvVars, fmt.Sprintf("ABT_THREAD_STACKSIZE=%d",
+					MIN_ABT_THREAD_STACKSIZE_FOR_DCPM))
+			} else {
+				stacksize_value, err := strconv.Atoi(stacksize_str)
+				if err == nil {
+					if stacksize_value < MIN_ABT_THREAD_STACKSIZE_FOR_DCPM {
+						return errors.New(fmt.Sprintf("env_var ABT_THREAD_STACKSIZE should be >= %d for 'dcpm' storage class instead of (%d)",
+							MIN_ABT_THREAD_STACKSIZE_FOR_DCPM, stacksize_value))
+					}
+				} else {
+					return errors.New(fmt.Sprintf("env_var ABT_THREAD_STACKSIZE has invalid value: %s",
+						stacksize_str))
+
+				}
+			}
+			// Ensure default handling of shutdown state (SDS) for DCPM storage class.
+			if pmemobj_conf_err == nil && strings.Contains(pmemobj_conf_str, "sds.at_create") {
+				return errors.New("env_var PMEMOBJ_CONF should NOT be set to sds.at_create=? for 'dcpm' storage class")
+			}
+		} else {
+			// Disable shutdown state (SDS) (part of RAS) for RAM-based simulated SCM.
+			// RAM doesn't support this feature and trying to use
+			// it will fail the create/open operations.
+			if pmemobj_conf_err != nil {
```

Comment:
go vars should be named in the form `pmemobjConfErr`

Reply:
Done